isolated-jobs-test/src/main/java/com/hazelcast/jet/tests/isolated/IsolatedJobsStreamTest.java

```diff
@@ -0,0 +1,170 @@
+    private int snapshotIntervalMs;
+    private int logVerificationCountThreshold;
+    private int sleepBeforeValidationInMinutes;
+    private int jobWindowTumbleInMinutes;
+    private int sleepBetweenStreamRecordsMs;
+
+    public static void main(final String[] args) throws Exception {
+        new IsolatedJobsStreamTest().run(args);
+    }
+
+    @Override
+    protected boolean runOnBothClusters() {
+        return true;
+    }
+
+    @Override
+    protected void init(HazelcastInstance client) throws Exception {
+        snapshotIntervalMs = propertyInt("snapshotIntervalMs",
+                DEFAULT_SNAPSHOT_INTERVAL_MS);
+        logVerificationCountThreshold = propertyInt("logVerificationCountThreshold",
+                DEFAULT_LOG_VERIFICATION_COUNT_THRESHOLD);
+        sleepBeforeValidationInMinutes = propertyInt("sleepBeforeValidationInMinutes",
+                DEFAULT_SLEEP_BETWEEN_VALIDATIONS_IN_MINUTES);
+        jobWindowTumbleInMinutes = propertyInt("jobWindowTumbleInMinutes",
+                DEFAULT_JOB_WINDOW_TUMBLE_IN_MINUTES);
+        sleepBetweenStreamRecordsMs = propertyInt("sleepBetweenStreamRecordsMs",
+                DEFAULT_SLEEP_BETWEEN_STREAM_RECORDS_MS);
+    }
+
+    @Override
+    protected void test(HazelcastInstance client, String name) throws Throwable {
+        long begin = System.currentTimeMillis();
+        String logPrefix = "[" + name + "] ";
+
+        UUID excludedMember = client.getCluster().getMembers().iterator().next().getUuid();
+        Job streamJob = JobDefinition.createStreamJob(jobWindowTumbleInMinutes, sleepBetweenStreamRecordsMs,
+                snapshotIntervalMs, client, excludedMember, name);
+        waitForJobStatus(streamJob, RUNNING);
+
+        long validationCount = 0;
+        while (System.currentTimeMillis() - begin < durationInMillis) {
+            if (getJobStatusWithRetry(streamJob) == JobStatus.FAILED) {
+                throw new RuntimeException(logPrefix + "Job failed: " + streamJob.getName());
+            }
+
+            sleepMinutes(sleepBeforeValidationInMinutes);
+
+            assertStreamSink(client, excludedMember, name);
+            clearSink(client, name);
+
+            validationCount++;
+            if (validationCount % logVerificationCountThreshold == 0) {
+                logger.info(logPrefix + "Validations count: " + validationCount);
+            }
+        }
+        logger.info(logPrefix + "Final validations count: " + validationCount);
+    }
+
+    @Override
+    protected void teardown(Throwable t) throws Exception {
+
+    }
+
+    private void clearSink(HazelcastInstance client, String jobName) {
+        client.getList(jobName).clear();
+    }
+
+    private void assertStreamSink(HazelcastInstance client, UUID excludedMember, String jobName) {
+        IList<List<UUID>> list = client.getList(jobName);
+        assertFalse("There was no records in the List from job " + jobName, list.isEmpty());
```

Comment:
This is incorrect. List is not persisted after full cluster restart => it will be empty after the restart and before the first job is executed.

Generally using List for this kind of test doesn't seem to be a good idea.

Reply:
I'm using map now.